internal/pkg/service/common/etcdop/watch.go

```diff
@@ -3,14 +3,23 @@ package etcdop
 import (
 	"context"
 
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
 	"go.etcd.io/etcd/api/v3/mvccpb"
 	etcd "go.etcd.io/etcd/client/v3"
 )
 
 type EventT[T any] struct {
-	Type  mvccpb.Event_EventType
-	Key   string
-	Value *T
+	*etcd.Event
+	Header *etcdserverpb.ResponseHeader
```

Comment:
2. Tu by bolo dobre uchovavat cely raw event a aj response header

Obsahuje to napr. `revision`, co pri niektorych atomickych operaciach/transakciach potrebujes.